### PR TITLE
Fix `Genesis` block hash calculation for `Testnet` network

### PR DESCRIPTION
**Pull request by m-Peter**
Closes: https://github.com/onflow/flow-evm-gateway/issues/534

## Description

1. The hash calculation for the `Genesis` block on `Testnet` should be done without the `PrevRandao` field, as it was introduced later, and it produces a different hash.

2. The `ParentHash` for the first EVM block that contains the `PrevRandao` field, is pointing to a non-existent block. Relevant events:
```bash
A.8c5303eaa26202d6.EVM.BlockExecuted(
    height: 1385490,
    hash: "0xfc9a0204b12bc4875ed880300d07f27de4f9857be67d13cf72f53b09e8942ced",
    timestamp: 1724362036,
    totalSupply: 37735203666010000000000,
    totalGasUsed: 0,
    parentHash: "0x95b9d3a30ce797062ab5162d6dc4c60d887ff66eeec49640931c2d7ac014974d",
    receiptRoot: [86, 232, 31, 23, 27, 204, 85, 166, 255, 131, 69, 230, 146, 192, 248, 110, 91, 72, 224, 27, 153, 108, 173, 192, 1, 98, 47, 181, 227, 99, 180, 33],
    transactionHashRoot: [86, 232, 31, 23, 27, 204, 85, 166, 255, 131, 69, 230, 146, 192, 248, 110, 91, 72, 224, 27, 153, 108, 173, 192, 1, 98, 47, 181, 227, 99, 180, 33]
)

A.8c5303eaa26202d6.EVM.BlockExecuted(
    height: 1385491,
    hash: "0x829f0aeea56301ca085a9688f0ab238f2738e553a09e00963170d92c08957daa",
    timestamp: 1724362037,
    totalSupply: 37735203666010000000000,
    totalGasUsed: 0,
    parentHash: "0x0e742970c53cc2924d55830a588609ca1db460cce41850df7af119639fe42d91",
    receiptRoot: [86, 232, 31, 23, 27, 204, 85, 166, 255, 131, 69, 230, 146, 192, 248, 110, 91, 72, 224, 27, 153, 108, 173, 192, 1, 98, 47, 181, 227, 99, 180, 33],
    transactionHashRoot: [86, 232, 31, 23, 27, 204, 85, 166, 255, 131, 69, 230, 146, 192, 248, 110, 91, 72, 224, 27, 153, 108, 173, 192, 1, 98, 47, 181, 227, 99, 180, 33],
    prevrandao: [84, 243, 47, 99, 190, 212, 43, 169, 197, 65, 166, 231, 112, 226, 218, 121, 153, 240, 149, 165, 22, 98, 37, 215, 150, 128, 215, 244, 192, 84, 175, 161]
)
```

**Note:** After deployment, a re-index is needed for this code to take effect.

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-nft/blob/master/CONTRIBUTING.md#styleguides).
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels 

## Summary by CodeRabbit

- **New Features**
	- Introduced specific logic for creating a Genesis block for the Testnet chain, ensuring it meets the unique requirements of this environment.
	- Enhanced block retrieval logic to handle specific block heights and update parent block hashes for Testnet.

- **Tests**
	- Added a new test to verify the hash of the Genesis block for Testnet, ensuring accuracy and reliability of the implementation.

### CODE REVIEW

**Review by ramtinms — APPROVED** [review]
Do we have the same problem for the mainnet?

**Comment by m-Peter** [comment]
> Do we have the same problem for the mainnet?

Nope, this is just a problem on `testnet`. On `mainnet`, there have not been any hash calculation breaking changes.

File: models/block.go [hunk 1]
```diff
@@ -80,6 +121,14 @@ func decodeBlockEvent(event cadence.Event) (*Block, error) {
 		return nil, fmt.Errorf("failed to cadence decode block [%s]: %w", event.String(), err)
 	}
 
+	if event.EventType.ID() == testnetEVMBlockExecutedEventID {
```
Comment (sideninja) [inline models/block.go:124]: I would move this to block decoding from database on the model, not on the decoding of events

**Comment by j1010001** [comment]
Needs an update from @m-Peter and then @ramtinms agrees we can merge

File: storage/pebble/blocks.go [hunk 2]
```diff
@@ -270,5 +276,26 @@ func (b *Blocks) getBlock(keyCode byte, key []byte) (*models.Block, error) {
...
+	if b.chainID == flowGo.Testnet && slices.Contains(testnetBrokenParentHashHeights, block.Height) {
+		data, err := b.store.get(blockHeightKey, uint64Bytes(block.Height-1))
+		if err != nil {
+			return nil, fmt.Errorf("failed to get block: %w", err)
+		}
+		parentBlock, err := models.NewBlockFromBytes(data)
+		if err != nil {
+			return nil, err
+		}
+		parentBlockHash, err := parentBlock.Hash()
```
Comment (zhangchiqing) [inline storage/pebble/blocks.go:293]: can we add some comments about why we need to do this?

**Review by zhangchiqing — APPROVED** [review]

File: storage/pebble/blocks.go (hunk 2 above)
Comment (m-Peter) [reply storage/pebble/blocks.go]: Good point :+1: I have added a descriptive comment in: https://github.com/onflow/flow-evm-gateway/pull/566/commits/f14ad0cb903abc7270d6ebbacda6e014d5a030ef

**Review by janezpodhostnik — APPROVED** [review]

**Pull request merged**